nixos/modules/system/activation/top-level.nix

```diff
@@ -59,7 +59,10 @@ let
       ${if config.boot.initrd.systemd.enable then ''
         cp ${config.system.build.bootStage2} $out/prepare-root
         substituteInPlace $out/prepare-root --subst-var-by systemConfig $out
-        ln -s "$systemd/lib/systemd/systemd" $out/init
+        # This must not be a sylink or the abs_path of the grub builder for the tests
+        # will resolve the symlink and we end up with a path that doesn't point to a
+        # system closure.
+        cp "$systemd/lib/systemd/systemd" $out/init
```

Comment:
Should we fix the test instead?

Reply:
That would touch the `init=` parameter in the grub builder, something I don't feel comfortable doing.
Also it's not unlikely that this would happen with real systems as well, I didn't check further